From 2759e8760d898519ff9f0087bb4908c13dda134b Mon Sep 17 00:00:00 2001
From: Christian Lamparter <chunkeey@googlemail.com>
Date: Thu, 28 Mar 2013 21:17:48 +0100
Subject: [PATCH 03/65] r92su: try to fix high byte vs low byte swap for big
 endian system
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

José Vázquez reported a issue with r92su. The usbmon
log he sent me did reveal that the address bytes are
swapped on big endian architectures.

(Big Endian)
Ci: s c0 05 0400 0000 0004 4 <
Ci: s c0 05 0a00 0000 0001 1 <

vs.

(Little Endian)
Ci: s c0 05 0004 0000 0004 4 <
Ci: s c0 05 000a 0000 0001 1 <

Reported-by: José Vázquez <ppvazquezfer@gmail.com>
Signed-off-by: Christian Lamparter <chunkeey@googlemail.com>
---
 r92su/usb.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

--- a/drivers/net/wireless/r92su/usb.c
+++ b/drivers/net/wireless/r92su/usb.c
@@ -88,7 +88,7 @@ static int r92su_sync_read(struct r92su
 	return (ret < 0) ? ret : ((ret == size) ? 0 : -EMSGSIZE);
 }
 
-static void r92su_read_helper(struct r92su *r92su, const u16 address,
+static void r92su_read_helper(struct r92su *r92su, const u32 address,
 			      void *data, const u16 size)
 {
 	int ret;
@@ -121,7 +121,7 @@ u32 r92su_read32(struct r92su *r92su, co
 	return le32_to_cpu(data);
 }
 
-static void r92su_write_helper(struct r92su *r92su, const u16 address,
+static void r92su_write_helper(struct r92su *r92su, const u32 address,
 			       const void *data, const u16 size)
 {
 	int ret;
